pkg/ccl/changefeedccl/sink_kafka.go

```diff
@@ -264,10 +267,16 @@ func (s *kafkaSink) Dial() error {
 		return err
 	}
 
-	if err = client.RefreshMetadata(s.Topics()...); err != nil {
-		// Now that we do not fetch metadata for all topics by default, we try
-		// RefreshMetadata manually to check for any connection error.
-		return errors.CombineErrors(err, client.Close())
+	// Make sure the broker can be reached.
+	broker := client.LeastLoadedBroker()
+	if broker != nil {
+		req := &sarama.ApiVersionsRequest{}
```

Comment:
the arguments for the ApiVersionsRequest are kinda sarama specific and i'd hate to have to replicate them.

instead i think you should do this (pseudocode):

```go
if err := broker.Open(s.kafkaCfg); err != nil && !errors.Is(err, sarama.AlreadyConnected) { .. }
if ok, err := broker.Connected(); !ok || err != nil { ... }
```

see https://pkg.go.dev/github.com/shopify/sarama#Broker.Open